core/src/main/scala/org/apache/spark/util/VersionUtils.scala

```diff
@@ -63,4 +64,37 @@ private[spark] object VersionUtils {
           s" version string, but it could not find the major and minor version numbers.")
     }
   }
+
+  /**
+   * Retrieves the major, minor and patch parts from the input `version`. Returns `None` if the
+   * input is not of a valid format.
```

Comment:
I think we should mention that minor/patch versions are filled in as 0 if they're not found. This is different from the behavior of other methods in this class (e.g. majorMinor will give an error if minor is not present)

Reply:
Yeah good point.